### PR TITLE
Check that functional tests cleanup their tables

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
This is an automated comment for commit de0837fe21d3f7330fc58712c887b1cc570af05a with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51991/de0837fe21d3f7330fc58712c887b1cc570af05a/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: tests/clickhouse-test [hunk 1]
```diff
@@ -1212,7 +1212,26 @@ class TestCase:
...
+                        "log_comment": args.testcase_basename,
+                    },
+                )
+                .decode()
+                .replace("\n", ", ")
+            )
+
+            if 0 != len(leftover_tables):
+                raise Exception(
+                    f"The test should cleanup its tables ({leftover_tables}), otherwise it is inconvenient for running it locally."
```
Comment (tavplubix) [inline tests/clickhouse-test:1231]: Why? `clickhouse-test` creates (and drops) a separate database for each test even if you run it locally (unless you explicitly specified `--database`)

Comment (Algunenano) [reply tests/clickhouse-test:1231]: +1 to this. Many test don't explicitly teardown their tables based on this fact too

Comment (alexey-milovidov) [reply tests/clickhouse-test:1231]: I run tests as follows:

```
clickhouse-client -n < test.sql
```
and
```
./test.sh
```

Comment (Algunenano) [reply tests/clickhouse-test]: Interesting. Out of curiosity, any reason not to use `./tests/clickhouse-test`?

Comment (alexey-milovidov) [reply tests/clickhouse-test]: Slightly less direct feedback, plus the need to change directories.

File: tests/clickhouse-test [hunk 2]
```diff
@@ -1212,7 +1212,29 @@ class TestCase:
             seconds_left = max(
                 args.timeout - (datetime.now() - start_time).total_seconds(), 20
             )
-            drop_database_query = "DROP DATABASE IF EXISTS " + database
+
+            # Check if the test does not cleanup its tables.
+            # Only for newly added tests. Please extend this check to the old tests as well.
+            if self.case_file >= "02800":
```
Comment (azat) [inline tests/clickhouse-test:1218]: case_file is absolute path, not a basename, so likely it never worked

**Pull request merged**